pkg/openstack/loadbalancer.go

```diff
@@ -2118,11 +2101,12 @@ func (lbaas *LbaasV2) ensureSecurityRule(
 	}
 
 	mc := metrics.NewMetricContext("security_group_rule", "create")
-	_, err = rules.Create(lbaas.network, sgRuleCreateOpts).Extract()
+	_, err := rules.Create(lbaas.network, sgRuleCreateOpts).Extract()
 	if mc.ObserveRequest(err) != nil {
```

Comment:
Note that by doing this we're generating an error in prometheus in all cases, even when the 'error' is expected as here. We have wrappers for this in CAPO: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/06f81e5b84c0fbeaed2bb71e8ec0ef6b36a90857/pkg/metrics/metrics.go#L56-L76

Reply:
> Note that by doing this we're generating an error in prometheus in all cases, even when the 'error' is expected as here. We have wrappers for this in CAPO: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/06f81e5b84c0fbeaed2bb71e8ec0ef6b36a90857/pkg/metrics/metrics.go#L56-L76

Ouch, you're right here, thanks!